pkg/vulnsrc/oracle-oval/oracle-oval.go

```diff
@@ -335,6 +356,30 @@ func (o *Oracle) Put(tx *bolt.Tx, input PutInput) error {
 	return nil
 }
 
+// removeVendorIDs removes VendorID from advisories + merges arches for advisories (by fixedVersion).
+// This is needed to save space in OSS trivy-db.
+// But Aqua storage requires this information.
+func removeVendorIDs(advs types.Advisories) types.Advisories {
+	versionToArches := make(map[string][]string) // fixed version -> arches
+	for _, entry := range advs.Entries {
+		versionToArches[entry.FixedVersion] = append(versionToArches[entry.FixedVersion], entry.Arches...)
```

Comment:
Isn't it possible that arches are duplicated?

Reply:
we can get duplicates only if we have 2 records with the same `fixed version` and `arch`.
But it is impossible because we store uniq arch+flavor:
https://github.com/aquasecurity/trivy-db/blob/47d4c6ca71d4261e331332b1209a09d780316765/pkg/vulnsrc/oracle-oval/oracle-oval.go#L268-L292

But if you found this case - give me an example and I will check.

Reply:
I'm probably missing something, but the logic you shared doesn't appear to help for me.

![image](https://github.com/user-attachments/assets/79c29c28-895a-4030-96cb-f9e2ef56d401)

Reply:
selectLatestVersions returns only 2 values:
- x86_64 + normal flavors - v1.0.0 + ELSA-0001
- arm64 + normal flavors - v1.0.0 + ELSA-0001

https://github.com/aquasecurity/trivy-db/blob/47d4c6ca71d4261e331332b1209a09d780316765/pkg/vulnsrc/oracle-oval/oracle-oval.go#L277-L289


This PR does not preserve all ELSA-ids. IIUC @santhosh1729 - he only needs one ELSA-ID (for the latest fixed version).

Reply:
I see. It's too complicated that different ELSA-IDs are used per arch. I'm sure it will cause bugs.

![image](https://github.com/user-attachments/assets/39a1dc25-e0a8-46dc-9fae-2067efbff8e2)

Reply:
Could we simplify implementation by treating the ELSA-ID as the primary ID, thereby eliminating the need for aggregation? It might be necessary to upgrade to Trivy DB v3, as the current Trivy client does not support it.

Reply:
> It's too complicated that different ELSA-IDs are used per arch. I'm sure it will cause bugs.

it all depends on how aqua plans to use it. With the right approach - I don't think there will be any problems with it. And this confusion is due to confusion in the oracle advisories.

> Could we simplify implementation by treating the ELSA-ID as the primary ID

this will simplify the work for us.
But on the other hand, users will have to handle the nuances of Oracle advisories themselves (e.g., when several ELSA-IDs fix one CVE-ID).

Reply:
>  It might be necessary to upgrade to Trivy DB v3

I don't mind. We already talked about switching to v3.

Reply:
>But on the other hand, users will have to handle the nuances of Oracle advisories themselves (e.g., when several ELSA-IDs fix one CVE-ID).

I think Trivy can convert the detected ELSA-ID into a CVE-ID and retrieve only the latest version if there are duplicate CVE-IDs. This way, we wouldn't have to worry about arch or flavor. Also, we can keep showing users the same information (CVE-IDs). What do you think?

Reply:
We do something like that for RedHat:
https://github.com/aquasecurity/trivy/blob/4202c4ba0d8fcff4b89499fe03050ef4efd37330/pkg/detector/ospkg/redhat/redhat.go#L117-L118

So we can use similar logic.

Main point is that we need to change the Trivy code.
And I also don't know how to change the size of DB.

Reply:
>Main point is that we need to change the Trivy code.

Yes, so we cannot apply the change now.

>And I also don't know how to change the size of DB.

Also, true. We need to check it.

Anyway, I'll merge this PR and think about v3 later.